[bug] - Using sysroot for 'iPhoneOS' but targeting 'MacOSX'

### Describe the bug

I'm attempting to port Tauri In App Payments to both v2 as well as iOS. Everything is building, but when running tauri ios dev i'm getting an invalid manifest error.

I had posted on Discord about if this was a swift-rs issue or a Tauri issue and was told to open here, but it's possibly a swift-rs issue.

### Reproduction

https://github.com/stolinski/scott-ttemp

Run `tauri dev ios`

### Expected behavior

App loads.

### Full `tauri info` output

```text
[✔] Environment
    - OS: Mac OS 15.0.0 arm64 (X64)
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.81.0 (eeb90cda1 2024-09-04)
    ✔ cargo: 1.81.0 (2dbb1af80 2024-08-20)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 20.16.0
    - pnpm: 9.7.0
    - npm: 10.8.1
    - bun: 1.0.25

[-] Packages
    - tauri 🦀: 2.0.0-rc.6
    - tauri-build 🦀: 2.0.0-rc.6
    - wry 🦀: 0.42.0
    - tao 🦀: 0.29.1
    - @tauri-apps/api : 2.0.0-rc.4
    - @tauri-apps/cli : 2.0.0-rc.13

[-] Plugins
    - tauri-plugin-shell 🦀: 2.0.0-rc.2
    - @tauri-apps/plugin-shell : 2.0.0-rc.1

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../build
    - devUrl: http://localhost:1420/
    - framework: Svelte
    - bundler: Vite

[-] iOS
    - Developer Teams: Level Up Tutorials LLC (ID: AS6YFWZTTJ), Scott Tolinski (ID: WX3RQ7F2B7)
```


### Stack trace

```text
error: failed to run custom build command for `tauri-plugin-iap v0.0.1 (/Users/scotttolinski/Sites/projects/decider-app/plugins/iap)`

Caused by:
  process didn't exit successfully: `/Users/scotttolinski/Sites/projects/decider-app/src-tauri/target/debug/build/tauri-plugin-iap-096f22ba769ec6f5/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-link-search=native=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos
  cargo:rustc-link-search=native=/usr/lib/swift
  cargo:rustc-link-lib=clang_rt.ios
  cargo:rustc-link-search=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/16/lib/darwin

  --- stderr
  error: 'swift-iap': Invalid manifest (compiled with: ["/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc", "-vfsoverlay", "/var/folders/fm/_sgpt7q91_x97jrvd53whjtm0000gn/T/TemporaryDirectory.0ewK2P/vfs.yaml", "-L", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI", "-lPackageDescription", "-Xlinker", "-rpath", "-Xlinker", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI", "-target", "arm64-apple-macosx13.0", "-sdk", "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk", "-F", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks", "-I", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib", "-L", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib", "-swift-version", "5", "-I", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI", "-sdk", "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk", "-package-description-version", "5.10.0", "/Users/scotttolinski/Sites/projects/decider-app/plugins/iap/swift-iap/Package.swift", "-o", "/var/folders/fm/_sgpt7q91_x97jrvd53whjtm0000gn/T/TemporaryDirectory.FtT0p1/swift-iap-manifest"])
  <unknown>:0: warning: using sysroot for 'iPhoneOS' but targeting 'MacOSX'
  <unknown>:0: error: unable to load standard library for target 'arm64-apple-macosx13.0'
  error: 'swift-iap': Invalid manifest (compiled with: ["/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc", "-vfsoverlay", "/var/folders/fm/_sgpt7q91_x97jrvd53whjtm0000gn/T/TemporaryDirectory.JpQjNZ/vfs.yaml", "-L", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI", "-lPackageDescription", "-Xlinker", "-rpath", "-Xlinker", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI", "-target", "arm64-apple-macosx13.0", "-sdk", "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk", "-F", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks", "-I", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib", "-L", "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib", "-swift-version", "5", "-I", "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI", "-sdk", "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk", "-package-description-version", "5.10.0", "/Users/scotttolinski/Sites/projects/decider-app/plugins/iap/swift-iap/Package.swift", "-o", "/var/folders/fm/_sgpt7q91_x97jrvd53whjtm0000gn/T/TemporaryDirectory.m4eIAY/swift-iap-manifest"])
  <unknown>:0: warning: using sysroot for 'iPhoneOS' but targeting 'MacOSX'
  <unknown>:0: error: unable to load standard library for target 'arm64-apple-macosx13.0'
  thread 'main' panicked at /Users/scotttolinski/.cargo/registry/src/index.crates.io-6f17d22bba15001f/swift-rs-1.0.7/src-rs/build.rs:281:17:
  Failed to compile swift package TauriIAP
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:665:5
     1: core::panicking::panic_fmt
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:74:14
     2: swift_rs::build::SwiftLinker::link
               at /Users/scotttolinski/.cargo/registry/src/index.crates.io-6f17d22bba15001f/swift-rs-1.0.7/src-rs/build.rs:281:17
     3: build_script_build::main
               at ./build.rs:6:5
     4: core::ops::function::FnOnce::call_once
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
    Error [tauri_cli_node] Failed to run `cargo build`: command ["cargo", "build", "--package", "decider-app", "--manifest-path", "/Users/scotttolinski/Sites/projects/decider-app/src-tauri/Cargo.toml", "--target", "aarch64-apple-ios", "--features", "tauri/rustls-tls", "--lib", "--no-default-features"] exited with code 101
 ELIFECYCLE  Command failed with exit code 1.

note: Run script build phase 'Build Rust Code' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'decider-app_iOS' from project 'decider-app')
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution Build\ Rust\ Code /Users/scotttolinski/Library/Developer/Xcode/DerivedData/decider-app-hepjwdpaecdpowcrsqiophscuzit/Build/Intermediates.noindex/decider-app.build/debug-iphoneos/decider-app_iOS.build/Script-BFE46E4EC9F9451DEBF6C466.sh (in target 'decider-app_iOS' from project 'decider-app')
	Building workspace decider-app with scheme decider-app_iOS and configuration debug
(2 failures)
    Error command ["xcodebuild"] exited with code 65
 ELIFECYCLE  Command failed with exit code 1.
```


### Additional context

With some adjustments it worked just fine for MacOS but never for iOS.

--- a/crates/tauri-cli/src/interface/rust.rs
+++ b/crates/tauri-cli/src/interface/rust.rs
@@ -147,17 +147,17 @@ impl Interface for Rust {
       manifest
     };
 
-    if let Some(minimum_system_version) = &config.bundle.macos.minimum_system_version {
-      std::env::set_var("MACOSX_DEPLOYMENT_TARGET", minimum_system_version);
-    }
-
     if let Some(target) = &target {
       if target.ends_with("ios") || target.ends_with("ios-sim") {
         std::env::set_var(
           "IPHONEOS_DEPLOYMENT_TARGET",
           &config.bundle.ios.minimum_system_version,
         );
+      } else if let Some(minimum_system_version) = &config.bundle.macos.minimum_system_version {
+        std::env::set_var("MACOSX_DEPLOYMENT_TARGET", minimum_system_version);
       }
+    } else if let Some(minimum_system_version) = &config.bundle.macos.minimum_system_version {
+      std::env::set_var("MACOSX_DEPLOYMENT_TARGET", minimum_system_version);
     }
 
     let app_settings = RustAppSettings::new(config, manifest, target)?;

--- a/crates/tauri-cli/src/interface/rust.rs
+++ b/crates/tauri-cli/src/interface/rust.rs
@@ -147,15 +147,14 @@ impl Interface for Rust {
       manifest
     };
 
-    if let Some(target) = &target {
-      if target.ends_with("ios") || target.ends_with("ios-sim") {
-        std::env::set_var(
-          "IPHONEOS_DEPLOYMENT_TARGET",
-          &config.bundle.ios.minimum_system_version,
-        );
-      } else if let Some(minimum_system_version) = &config.bundle.macos.minimum_system_version {
-        std::env::set_var("MACOSX_DEPLOYMENT_TARGET", minimum_system_version);
-      }
+    let target_ios = target.as_ref().map_or(false, |target| {
+      target.ends_with("ios") || target.ends_with("ios-sim")
+    });
+    if target_ios {
+      std::env::set_var(
+        "IPHONEOS_DEPLOYMENT_TARGET",
+        &config.bundle.ios.minimum_system_version,
+      );
     } else if let Some(minimum_system_version) = &config.bundle.macos.minimum_system_version {
       std::env::set_var("MACOSX_DEPLOYMENT_TARGET", minimum_system_version);
     }

